docs: move vite docs from root to vite package (#294)
antfu/unocss: README.md

  ]
}
```

### Vite + Web Components

Since `Web Components` uses `Shadow DOM`, there is no way to style content directly from a global stylesheet (you can use `custom css vars` since those will penetrate the `Shadow DOM`) and so you only need to change the mode of the `UnoCss` plugin to `shadow-dom`.

With the `shadow-dom` mode, the `Unocss` plugin will scope all css styles declared on each individual web component to its own stylesheet, and so you only need to include `@unocss-placeholder` magic placeholder to inject generated css into the corresponding style css block.

```ts
// vite.config.js
import Unocss from 'unocss/vite'

export default {
  plugins: [
    Unocss({
      mode: 'shadow-dom',
      /* more options */
    }),
  ]
}
```

On each `web component` just add `@unocss-placeholder` to its style css block:
```ts
const template = document.createElement('template')
template.innerHTML = `
<style>
:host {...}
@unocss-placeholder
</style>
<div class="m-1em">
...
</div>
`
```

or using [Lit](https://lit.dev/):

```ts
@customElement('my-element')
export class MyElement extends LitElement {
  static styles = css`
    :host {...}
    @unocss-placeholder
  `
  ...
}
```

> #### `::part` built-in support

Reviewer: We should move this section to preset-mini or preset-uno, with a link here.

> You can use `::part` since `UnoCss` supports it via `shortcuts` and using `part-[<part-name>]:<shortcut>` rules.

Reviewer: ```suggestion
You can use `::part` is supported via `shortcuts` and using `part-[<part-name>]:<shortcut>` rules.
```

Reviewer: What is ::part tho? Can you share a link with me?

Author: https://meowni.ca/posts/part-theme-explainer/

Author: `::theme` has been removed from the proposal.

> `UnoCss` plugin uses `nth-of-type` to avoid collisions with multiple parts in the same web component and for the same parts on distinct web components, you don't need to worry about it, the plugin will take care for you.

Reviewer: ```suggestion
The default preset uses `nth-of-type` to avoid collisions with multiple parts in the same web component and for the same parts on distinct web components, you don't need to worry about it, the plugin will take care for you.
```
